### PR TITLE
Rename things to match desired terminology

### LINKED ISSUE #16: Renaming different things
In DeviceInterface.ts
Device => Thing
Interface => Schema

(all methods and properties e.g.)
DeviceCharacteristic
DeviceInterface
...

In DeviceAccessor.ts
DeviceAccessor => Thing or ThingDo

(all methods and properties e.g.)
getInterfaceAsync
createTranslatorAsync
getPropertyAsync
setPropertyAsync
addPropertyListener
removePropertyListener
invokeMethodAsync
getTranslatorInterface

In ITranslator.ts
as => resolveSchema

In AllJoynConverter.ts

AllJoynConverter => AllJoynSchemaReaderWriter

In PackageInfo.ts
Package => Manifest 
(Becomes ManifestInfo.ts)

Finally, for PackageSource.ts, PackageSourceUnion.ts please discuss with @jeanpa and figure out between the two of you where you want to end up on naming please.

### PR DESCRIPTION
**Pull request by jasongin**
This fixes #16 by updating type names, member names, parameter names, variable names, comments, and tests to match the terminology specified in that issue.

This PR contains _only renaming_; there are _no design changes_ here. In some places I have updated code or tests which will be refactored or eliminated as part of the upcoming design changes. Please do not get hung up on that! I wanted to be able to verify all unit tests were passing after this intermediate step, before I start breaking stuff for the design changes.

There are 2 deviations from the names specified in the issue:
1. The "Converter" classes have been split into separate "SchemaReader" and "SchemaWriter" classes. I did this because the classes can be relatively complex, while the reading and writing parts of them don't generally share code. And most of the time you'd only need one or the other.
2. The `PackageInfo` class is not renamed to `ManifestInfo` as suggested. For now I'm thinking that name should stay, because it is about more than just what is in the manifest.xml for a translator: rather it includes information about all kinds of OpenT2T modules in a package, including schemas, translators, and onboarding modules. This information will be critical for the functionality to dynamically download packages and load modules from them.

Regarding the `PackageSource` family of classes mentioned in the issue, I don't know of any reason those names should change.

### CODE REVIEW

**Comment by tjaffri** [comment]
Thanks! I'll review and likely merge tomorrow. Rock and roll.

File: node/lib/ThingAccessor.ts [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
+import { IThingTranslator } from "./IThingTranslator";
 
 import { EventEmitter } from "events";
 
 /**
- * Provides reflection-style access to device properties and methods via device interfaces.
+ * Provides reflection-style access to thing properties and methods via translators.
  */
-export class DeviceAccessor {
+export class ThingAccessor {
```
Comment (tjaffri) [inline node/lib/ThingAccessor.ts:10]: @jeanpa thinks we need to discuss this name further. I will open a separate issue, does not block this PR for now.

**Comment by tjaffri** [comment]
:shipit:

File: node/lib/ThingAccessor.ts (hunk 1 above)
Comment (jasongin) [inline node/lib/ThingAccessor.ts:10]: Sure, I didn't love the name either (before or after the change), but I didn't think of anything better.

**Comment by sanjaiganesh** [comment]
ah sorry couldn't get to review this. I shall look out for the design change one.

**Pull request merged**